src/main/java/com/TheAccountant/service/impl/LoanNotificationService.java

```diff
@@ -0,0 +1,59 @@
+package com.TheAccountant.service.impl;
+
+import com.TheAccountant.dao.LoanDao;
+import com.TheAccountant.dao.NotificationDao;
+import com.TheAccountant.model.loan.Loan;
+import com.TheAccountant.model.notification.Notification;
+import com.TheAccountant.model.notification.NotificationCategory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+/**
+ * Created by tudor.grigoriu on 3/19/2017.
+ */
+public class LoanNotificationService {
+
+    private static final String PREFIX = "Loan active! ";
+    private static final String TO_RECEIVE_FROM = "To receive from ";
+    private static final String TO_GIVE_TO = "To give to ";
+    @Autowired
+    private LoanDao loanDao;
+
+    @Autowired
+    private NotificationDao notificationDao;
+
+    @Scheduled(cron = "0 0 0 * * *") //everyday at midnight
+    public void addLoanNotifications(){
+        Timestamp currentTimestamp = new Timestamp(System.currentTimeMillis());
+        List<Loan> loanList = loanDao.findAllActiveBeforeDate(currentTimestamp);
```

Comment:
And also we can create a MEDIUM priority notification if there are only 3 days left from now until the date for the field 'until date' is reached. Or another period for Loan medium configuration that should be configured in application.properties.